service/controller/resource/clusternamespace/create.go

```diff
@@ -0,0 +1,47 @@
+package clusternamespace
+
+import (
+	"context"
+
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-apps-operator/pkg/project"
+	"github.com/giantswarm/cluster-apps-operator/service/controller/key"
+)
+
+// EnsureCreated creates the namespace for storing app related resources if it
+// does not exist.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.ClusterID(&cr),
+			Labels: map[string]string{
+				label.Cluster:   key.ClusterID(&cr),
+				label.ManagedBy: project.Name(),
```

Comment:
If this label includes a version such as `cluster-app-operator-1.1.5`, we can't actually update this from line number 36 since it's `create`. Can we handle this if it's true?

Reply:
This is the same way we create the giantswarm namespace in workload clusters.

https://github.com/giantswarm/app-operator/blob/77125cf33524c0900e5f00ed66d5e876c299b39e/service/controller/app/resource/tcnamespace/create.go#L65

So far we don't have a reason to update the labels. We just need to ensure that the namespace exists.